### ISSUE TITLE
worse visualizing in output !!!

### ISSUE THREAD

**Issue by farhadi76m** [jan-ackermann/Maskomaly#2]
Hello, I ran it but I didn't get good result ! 
Have you faced like this before ? 

![Untitled](https://github.com/jan-ackermann/Maskomaly/assets/47895920/cb04bee8-fd3d-40bc-8d78-2ec53915de1e)
![Untitled](https://github.com/jan-ackermann/Maskomaly/assets/47895920/89349e37-99f2-4128-85ff-8805ef12d3ed)

**Comment by jan-ackermann** [comment]
Hello, I have not encountered this problem. The only Issue that once appeared was related to unix file rights when executing the docker on a gpu cluster without sudo. Could you please try out the Docker Image to generate predictions? If that does not work, could you visualize the predictions of Mask2Former on the deer image (first in smiyc val.) and send me the indices of predictions that show the outlier.
I personally used CUDA 11.7 and Ubuntu 22.04.

**Comment by farhadi76m** [comment]
in this file you defined  `self.cp` twice !!! what is it ? 
moreover , I visualized the segmentation output, at first I found that my pretrained model was wrong but I changed it and I got correct result , then I ran the `compute_ood.py` but nothing changed ! 
it's the deer : 
![Untitled](https://github.com/jan-ackermann/Maskomaly/assets/47895920/ca35f13e-ef02-4c27-b04d-1b2ee1d8ea1a)

![Untitled](https://github.com/jan-ackermann/Maskomaly/assets/47895920/f496a515-05eb-4b68-8e50-af7d0140d09b)

also check it :
```
class Maskomaly(BaseSegmentationModel):
    def __init__(self, args):
        super().__init__(args)
        if args.analysis_file:
            self.cp = np.load(args.analysis_file)["cp"]
            self.ranking = np.argsort(self.cp)[::-1]
            self.cp = self.cp[self.ranking]
            self.take = int(args.masks) | np.argmax(self.cp < 0.25)
            self.ranking = self.ranking[:self.take]
        else:
            self.ranking = [49, 31, 83, 32]

        self.class_stats = np.zeros(10)
        self.pred_stats = np.zeros(10)
```

**Comment by farhadi76m** [comment]
sorry, I changed the `args.masks` and I got different outputs ! what is it ? 

if masks =1 : 
![Untitled](https://github.com/jan-ackermann/Maskomaly/assets/47895920/c07dc205-6c0c-4508-958c-48dbf6d09548)

if 100 : 
![Untitled](https://github.com/jan-ackermann/Maskomaly/assets/47895920/0c85e13a-4dde-46f3-bbe8-d187d0922254)

**Comment by jan-ackermann** [comment]
Hello,
cp are the mIoUs of the individual masks with the Anomalies in the SMIYC validation dataset.
args.mask is the number of masks that are used based on cp. 

Could you please provide more complete information on how you got these results?
Like, what call did you make, did you compute the mIoUs yourself, or are you using the standard indices, etc.

**Comment by farhadi76m** [comment]
I can't run the docker file .
sorry could you tell me witch pre-trained model did you  use ? or share me the checkpoint ?

**Comment by farhadi76m** [comment]
I got this result for 10 images , I ran the code in google colab . 
16.432%
51.15%
88.36%
16.43%
The average precision is: 16.432%
The auroc is: 51.15%
The fpr is: 88.36%
The aupr is: 16.43%

**Comment by farhadi76m** [comment]
I just ran it in step by step in colab jupyter manually and I got this scores!!! amazing !!! but it's also strange because when I'm taking to run it with bash command , I got wrong result ! 
![image](https://github.com/jan-ackermann/Maskomaly/assets/47895920/9e6bb376-c24b-49c2-9515-d58e48531497)

**Comment by jan-ackermann** [comment]
That is strange, but thank you for finding out! Would it be possible that you share this notebook?

**Comment by farhadi76m** [comment]
sure , first let me I simplified it and rewrite it then I'll send you an email .
